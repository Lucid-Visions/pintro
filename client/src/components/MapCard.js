import React, { Component } from "react";
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TouchableOpacity,
  Image
} from "react-native";
import ActionButtonComponent from "../components/ActionButton";

class MapCard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      user: props.user,
    };
  }
  render() {
    if(this.props.user == null){
      return null
    }
    return (
      <TouchableOpacity onPress={()=> this.props.navigation?this.props.navigation.navigate("Profile", {uid:this.props.user._id}):null}>
        <View style={styles.container}>
          <View style={styles.horizontalContainer}>
            <Image
              style={styles.profilePicture}
              source={{uri:this.props.user.profile_picture} || require("../assets/empty-profile-picture.png")}
            />
            <View style={styles.textContainer}>
              <Text style={styles.headingOne}>{this.props.user.name}</Text>
              <Text style={styles.headingTwo}>{this.props.user.experience.currentJobTitle}</Text>
              <Text style={styles.headingTwo}>{this.props.user.experience.currentCompany}</Text>
            </View>
          </View>
          <Text style={styles.descriptionText}>
            {this.props.user.bio.substring(0, 100)+"..."}
          </Text>
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#FBFBFB",
    width: 300,
    height: 150,
    borderRadius: 17,
    marginLeft: 10,
    marginRight: 10,
    marginTop: 10
  },
  horizontalContainer: {
    flexDirection: "row",
    alignContent: "center"
  },
  profilePicture: {
    height: 55,
    width: 55,
    borderRadius: 100,
    margin: 20
  },
  headingOne: {
    fontFamily: "poppins-medium",
    fontSize: 15
  },
  headingTwo: {
    fontFamily: "poppins-light",
    fontSize: 10
  },
  textContainer: {
    marginTop: 25
  },
  descriptionText: {
    fontFamily: "poppins-light",
    fontSize: 10,
    paddingLeft: 20,
    paddingRight: 20
  }
});

export default MapCard;
