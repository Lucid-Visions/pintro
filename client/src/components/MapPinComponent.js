import React, { Component } from "react";
import { Image, View, StyleSheet } from "react-native";

class MapPinComponent extends Component {
  constructor(props) {
    super(props);

  }

  getIcon() {
    switch (this.props.left) {
      case true: {
        switch (this.props.selected) {
          case true: {
            return require("../assets/leftMapIconYellow.png");
          }
          case false: {
            return require("../assets/leftMapIcon.png");
          }
        }
      }
      case false: {
        switch (this.props.selected) {
          case true: {
            return require("../assets/rightMapIconYellow.png");
          }
          case false: {
            return require("../assets/rightMapIcon.png");
          }
        }
      }
      default: {
        return require("../assets/rightMapIcon.png");
      }
    }
  }

  shouldComponentUpdate(nextProps){
    return(nextProps.insideIcon != this.props.insideIcon || nextProps.left != this.props.left || nextProps.selected != this.props.selected)

  }

  render() {
    return (
      <View style={styles.container}>
        <Image
          source={this.props.insideIcon?this.props.insideIcon:require("../assets/PeopleIconEllipse.png")}
          style={styles.insideIcon}
        />
        <Image source={this.getIcon()} style={styles.icon} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    height: 50
  },
  icon: {
    height: 45,
    resizeMode: "contain",
    aspectRatio: 1
  },
  insideIcon: {
    position: "absolute",
    zIndex: 99,
    height: 30,
    width: 30,
    marginTop: 7,
    marginLeft: 3,
    resizeMode: "contain",
    borderRadius: 50
  }
});
export default MapPinComponent;
