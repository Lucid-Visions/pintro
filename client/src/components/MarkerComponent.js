import MapView, { Marker } from "react-native-maps";
import React, { Component } from "react";
import MapPinComponent from "./MapPinComponent";

class MarkerComponent extends Component {
  constructor(props) {
    super(props);
    this.state = {
      id: props.id,
      coordinate: props.coordinate,
      index: props.index,
      selected:props.selected
    };
  }

  shouldComponentUpdate(nextProps){
      return (this.props.coordinate != nextProps.coordinate || this.props.selected != nextProps.selected || this.props.index != nextProps.index)
  }
  render() {
    return (
      <Marker
        key={this.props.id}
        coordinate={this.props.coordinate}
        onPress={() => {
          this.props.scrollToIndex(this.props.index);
        }}
      >
        <MapPinComponent selected={this.props.index == this.props.selected} left={false} />
      </Marker>
    );
  }
}

export default MarkerComponent;
