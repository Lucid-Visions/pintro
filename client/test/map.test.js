import React from "react";
import MapCard from "../src/components/MapCard";
import MarkerComponent from "../src/components/MarkerComponent"
import MapPinComponent from "../src/components/MapPinComponent"
import {create, act} from "react-test-renderer";

/**
 * tests for the MapCard component
 */

describe("<MapCard />", () => {
  // set up props
  const createTestProps = (props) => ({
    navigation: {
      navigate: jest.fn(),
    },
    ...props,
  });

  let props;
  beforeEach(() => {
    props = createTestProps({
      user: {
        uid: 123,
        name: "Kim Berguoni",
        bio:
          "Lorem ipsum dolor sit amet," +
          "consectetur adipiscing elit, sed do eiusmod ",
        experience: {
          currentJobTitle: "Window cleaner",
          currentCompany: "CleanWindows Ltd.",
        },
      },
    });
  });

  test("test if MapCard renders properly", () => {
    const mapCard = create(<MapCard {...props} />).toJSON();
    expect(mapCard).toMatchSnapshot();
  });

  test("test if empty MapCard will render null", () => {
    const mapCard = create(<MapCard />).toJSON();
    expect(mapCard).toMatchSnapshot();
  });

  test("test when you click on MapCard to navigate to Profile with the given uid in props", () => {
    const mapCard = create(<MapCard {...props} />);
    const instance = mapCard.root;
    const cardView = instance.findByType("View")

    cardView.props.onClick()
    expect(props.navigation.navigate).toHaveBeenCalledWith("Profile",{uid: props.uid})
  })
});

/**
 * Tests for the markers on the map
 */
describe("<MarkerComponent />", () => {
  let index;
  const scrollMock = jest.fn((i) => index = i)
  const props = {
    id: 1,
    index: 22,
    coordinate: {
      latitude: 0.12123231,
      longitude: 0.002331232
    },
  }

  test("snapshot of component", () => {
    const component = create(<MarkerComponent scrollToIndex={scrollMock} {...props}/>).toJSON();
    expect(component).toMatchSnapshot();
  })

  test("test if when you click on marker it brings you to the correct index", () => {
    const component = create(<MarkerComponent scrollToIndex={scrollMock} {...props}/>);
    const instance = component.root;
    const marker = instance.findByType("AIRMapMarker");
    
    expect(index).toBeFalsy()
    marker.props.onPress()
    expect(props.index).toBe(index)
  })

  test("snapshots of all variations of <MapPinComponent /> used in <MarkerComponent />", () => {
    const selectedRight = create(<MapPinComponent selected={true} left={false}/>).toJSON();
    const selectedLeft = create(<MapPinComponent selected={true} left={true}/>).toJSON();

    const notSelectedLeft = create(<MapPinComponent selected={false} left={true}/>).toJSON();

    const defaultComponent = create(<MapPinComponent />).toJSON();

    expect(selectedRight).toMatchSnapshot();
    expect(selectedLeft).toMatchSnapshot();
    expect(notSelectedLeft).toMatchSnapshot();
    expect(defaultComponent).toMatchSnapshot();
  })

})


