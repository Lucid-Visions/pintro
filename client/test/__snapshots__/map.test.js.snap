// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapCard /> test if MapCard renders properly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FBFBFB",
        "borderRadius": 17,
        "height": 150,
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 10,
        "width": 300,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "url": undefined,
          }
        }
        style={
          Object {
            "borderRadius": 100,
            "height": 55,
            "margin": 20,
            "width": 55,
          }
        }
      />
      <View
        style={
          Object {
            "marginTop": 25,
          }
        }
      >
        <Text
          style={
            Object {
              "fontFamily": "poppins-medium",
              "fontSize": 15,
            }
          }
        >
          Kim Berguoni
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "poppins-light",
              "fontSize": 10,
            }
          }
        >
          Window cleaner
        </Text>
        <Text
          style={
            Object {
              "fontFamily": "poppins-light",
              "fontSize": 10,
            }
          }
        >
          CleanWindows Ltd.
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "fontFamily": "poppins-light",
          "fontSize": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
        }
      }
    >
      Lorem ipsum dolor sit amet,consectetur adipiscing elit, sed do eiusmod ...
    </Text>
  </View>
</View>
`;

exports[`<MapCard /> test if empty MapCard will render null 1`] = `null`;

exports[`<MarkerComponent /> snapshot of component 1`] = `
<AIRMapMarker
  coordinate={
    Object {
      "latitude": 0.12123231,
      "longitude": 0.002331232,
    }
  }
  onPress={[Function]}
  stopPropagation={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "absolute",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "height": 50,
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "borderRadius": 50,
          "height": 30,
          "marginLeft": 3,
          "marginTop": 7,
          "position": "absolute",
          "resizeMode": "contain",
          "width": 30,
          "zIndex": 99,
        }
      }
    />
    <Image
      source={1}
      style={
        Object {
          "aspectRatio": 1,
          "height": 45,
          "resizeMode": "contain",
        }
      }
    />
  </View>
</AIRMapMarker>
`;

exports[`<MarkerComponent /> snapshots of all variations of <MapPinComponent /> used in <MarkerComponent /> 1`] = `
<View
  style={
    Object {
      "height": 50,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 50,
        "height": 30,
        "marginLeft": 3,
        "marginTop": 7,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 30,
        "zIndex": 99,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "aspectRatio": 1,
        "height": 45,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;

exports[`<MarkerComponent /> snapshots of all variations of <MapPinComponent /> used in <MarkerComponent /> 2`] = `
<View
  style={
    Object {
      "height": 50,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 50,
        "height": 30,
        "marginLeft": 3,
        "marginTop": 7,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 30,
        "zIndex": 99,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "aspectRatio": 1,
        "height": 45,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;

exports[`<MarkerComponent /> snapshots of all variations of <MapPinComponent /> used in <MarkerComponent /> 3`] = `
<View
  style={
    Object {
      "height": 50,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 50,
        "height": 30,
        "marginLeft": 3,
        "marginTop": 7,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 30,
        "zIndex": 99,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "aspectRatio": 1,
        "height": 45,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;

exports[`<MarkerComponent /> snapshots of all variations of <MapPinComponent /> used in <MarkerComponent /> 4`] = `
<View
  style={
    Object {
      "height": 50,
    }
  }
>
  <Image
    source={1}
    style={
      Object {
        "borderRadius": 50,
        "height": 30,
        "marginLeft": 3,
        "marginTop": 7,
        "position": "absolute",
        "resizeMode": "contain",
        "width": 30,
        "zIndex": 99,
      }
    }
  />
  <Image
    source={1}
    style={
      Object {
        "aspectRatio": 1,
        "height": 45,
        "resizeMode": "contain",
      }
    }
  />
</View>
`;
